tests/pubsub/testgossipsub.nim

```diff
@@ -220,6 +220,70 @@ suite "GossipSub":
 
     await allFuturesThrowing(nodesFut.concat())
 
+  asyncTest "GossipSub's observers should run after message is sent, received and validated":
+    var
+      handlerFut = newFuture[bool]()
+      recvCounter = 0
+      sendCounter = 0
+      validatedCounter = 0
+
+    proc handler(topic: string, data: seq[byte]) {.async.} =
+      check topic == "foo"
+      handlerFut.complete(true)
```

Comment:
Is this necessary?

Reply:
Not really, it was just an extra sanity check. But I can get rid of it if you want

Reply:
done